### ISSUE TITLE
Support for a config file so a single instance can match/exec multiple components

### ISSUE THREAD

**Issue by bradleybeddoes** [mattgreen/watchexec#33]
Thanks for the great work on watchexec.

Just wanted to leave a future feature request to support loading matches from a config file, essentially the way Guard works (https://github.com/guard/guard/wiki/Guardfile-examples).

This would mean a single instance of watchexec could handle multiple changes that might go on within a single development project, i.e. running tests on source code and compilation on scss, and neatly combine all output in a single terminal.

**Comment by mattgreen** [comment]
Great idea!

This will involve some pretty major changes, so I'm tagging it with 2.0. I'm open to discussion on what the config file will look like.

**Comment by bradleybeddoes** [comment]
Agreed, keep it as simple as possible. TOML seems like a reasonable choice, plain JSON might also work out.

Here is a basic example of what I had in mind, allows you to watch a globset and ignore specific matches within it and runs a simple command.

https://gist.github.com/bradleybeddoes/35c54b76a3b6b9986985f295ec28cc47

**Comment by Mange** [comment]
IMO, JSON is a bad config format for humans. The syntax is very strict, and it's meant for machines and communication over plain text channels with *readable* payloads. Writing JSON by hand is error prone (quote all keys, trailing commas are invalid, etc). 

I'd recommend YAML over JSON as it's more forgiving.

**Comment by passcod** [comment]
Reclassifying as I don't think it's as much of a change now as it was four years ago, and could be done even without a breaking change.

**Comment by passcod** [comment]
Being worked on in #234 and #235.
